app/controllers/objects_controller.rb

```diff
@@ -9,57 +9,69 @@ class ObjectsController < ApplicationController
   # return the PreservedObject model for the druid (supplied with druid: prefix)
   # GET /objects/:druid
   def show
-    object = PreservedObject.find_by(druid: strip_druid(params[:id]))
-    if object
-      render json: object.to_json
-    else
-      render status: 404, json: 'Object not found'
-    end
+    render json: PreservedObject.find_by!(druid: druid).to_json
   end
 
   # return the checksums and filesize for a single druid (supplied with druid: prefix)
   # GET /objects/:druid/checksum
   def checksum
-    render json: checksum_for_object(params[:id]).to_json
-  rescue Moab::ObjectNotFoundException => e
-    render status: 404, json: e.message
-  rescue StandardError => e
-    render status: 500, json: e.message
+    render json: checksum_for_object(druid).to_json
   end
 
   # return the checksums and filesize for a list of druid (supplied with druid: prefix)
   # GET /objects/checksums?druids=druida,druidb,druidc
   def checksums
-    druids = params[:druids]
+    unless druids.present?
+      render(plain: '400 bad request', status: :bad_request)
+      return
+    end
+
     respond_to do |format|
       format.json do
-        results = druids.map { |druid| { "#{druid}": checksum_for_object(druid) } }
-        render json: results.to_json
+        render json: json_checksum_list
       end
       format.csv do
-        results = CSV.generate do |csv|
-          druids.each do |druid|
-            checksum_for_object(druid).each do |checksum|
-              csv << [druid, checksum[:filename], checksum[:md5], checksum[:sha1], checksum[:sha256], checksum[:filesize]]
-            end
-          end
-        end
-        render plain: results
+        render plain: csv_checksum_list
       end
-      format.any  { render status: 406, plain: 'Format not acceptable' }
+      format.any  { render status: :not_acceptable, plain: 'Format not acceptable' }
     end
-  rescue Moab::ObjectNotFoundException => e
-    render status: 404, json: e.message
-  rescue StandardError => e
-    render status: 500, json: e.message
   end
 
   private
 
+  def druid
+    strip_druid(params[:id])
+  end
+
+  def druids
+    return [] unless params[:druids].present?
+    params[:druids].map { |druid| strip_druid(druid) }.sort.uniq # normalize, then sort, then de-dupe
+  end
```

Comment:
tried to centralize all the param wrangling to make the public controller methods more readable, and more consistent in their input handling.